eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -161,35 +170,42 @@ object Graph {
       }
 
       /**
-        * Adds and edge to the graph, if one of the two vertices is not found, it will be created
+        * Adds and edge to the graph, if one of the two vertices is not found, it will be created.
+        * Edges are added as "reversed" swapping the starting vertex with the final.
         *
         * @param edge the edge that is going to be added to the graph
         * @return a new graph containing this edge
         */
       def addEdge(edge: GraphEdge): DirectedGraph = {
 
-        val vertexIn = edge.desc.a
-        val vertexOut = edge.desc.b
+        val finalEdge = edge.copy(desc = reverseDesc(edge.desc))
```

Comment:
how significant is this extra copy during graph initialization on mobile?

Reply:
We don't call this method during the graph initialization, we use DirectedGraph.makeGraph...which also makes a copy of the edge desc